plugins/kubernetes-backend/examples/dice-roller/component.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: dice-roller
+  description: It rolls dice
+  tags:
+    - go
+  annotations:
+    'backstage.io/kubernetes-id': dice-roller
```

Comment:
I just wish there was some way to not have to have this unique extra annotation to add to every object we want to relate to. Maybe there is no way, I am in deep water here so I don't have any real opinions except feeling that it's unfortunate.

Could we somehow mirror the selections that the objects in the manifest do? Like, essentially saying that we want to match on `label.app=dice-roller` and `spec.selector.matchLabels.app=dice-roller` or something :) Not sure where that would place things like ConfigMap and Secret, but seems to me that they are kind of weakly related and therefore in need of something like this special annotation ...

Reply:
Similar to what I said in https://github.com/spotify/backstage/pull/2571#discussion_r494128339

I think this is an easy way to set it up for now, but I don't think it has to be set in stone like this. We could add more ways to discover related kubernetes objects later. Such as returning everything in a namespace.

Reply:
Cool. Yeah let's discuss that separately at some point, because it does sound a lot easier to my ears, if the Backstage side was the only one that had to declare labels/annotations/selectors/whatever mechanism without having to do anything to the k8s side

Reply:
Yeah, another option could be to specify the label key/value to look for. Then for example we might be able to use tugboat labels.